scripts/update-generator/main.go

```diff
@@ -129,10 +147,23 @@ func createUpdate(targetPath string, channel, version, platform string, installe
 		return errs.Wrap(err, "Could not create temp binary dir")
 	}
 
-	for _, bf := range binaries {
-		err := copyFileToDir(bf, binTempDir, true)
+	for _, f := range files {
+		err := copyFileToDir(f, binTempDir, true)
 		if err != nil {
-			return errs.Wrap(err, "Failed to copy binary file %s", bf)
+			return errs.Wrap(err, "Failed to copy binary file %s", f)
+		}
+	}
+
+	if systemAppDir != "" {
+		sysTempDir := filepath.Join(tempDir, "system")
```

Comment:
use `ioutil.TempDir()`

Reply:
The installer expects a directory named `system` in its archive. I don't think that `ioutil.TempDir()` can do that as it appends a random string to the end of the filename

Reply:
you can still use `ioutil.TempDir()` for the value of `tempDir`. My worry with the current approach is that the value is constant, so you could run into conflicts.

Reply:
Fair enough, I doubled checked and `tempDir` is using `ioutil.TempDir()` https://github.com/ActiveState/cli/blob/master/scripts/update-generator/main.go#L110